### PR TITLE
Issue490 add manual export

### PR DESCRIPTION
**Pull request by eeverman**
### All Submissions:

Have you checked that...
* [x] Your pull request is to the *_main_* branch
* [x] Your code is up to date with the latest code from the *_main_*
* [x] You followed the guidelines in our [Developer Guide](https://sites.google.com/view/andhow/developer)?
* [x] Your code does not decrease test coverage (maybe it improves coverage!!)
* [x] If this is related to an issue, please include a link to the issue with the 'Fixes #XXX' syntax.

### CODE REVIEW

**Comment by eeverman** [comment]
Fixes #490

**Comment by eeverman** [comment]
Hey @alex-kar - This is a major new feature addition that is ready for review:  Manual Exports.
The place to see it used is `andhow-testing/andhow-simulated-app-tests/example-app-3`
### The basic idea
In your application, you can use AndHow Properties for configuration.  This gives you validation, ability to configure from any/multiple sources, detailed error messages, 'fail-fast' for bad configuration, etc..  But, frameworks like Hibernate, MyBatis and others don't expect to find their configuration in AndHow Properties.

How can you use AndHow for configuration and integrate with other frameworks?

A Manual Export provides a way to do your configuration in AndHow, then export select Properties to a Map or `java.util.Propertiers` for use by frameworks that take their configuration in this way.

This is a fairly large addition to AndHow - I hope you have some time to dig into a bit.  I hope you have some questions and we can talk about it.  I'm curious to hear your thoughts on:  Is this the best way to do it?  How useful would this be?

File: andhow-core/src/main/java/org/yarnandtail/andhow/api/StaticPropertyConfiguration.java [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package org.yarnandtail.andhow.api;
 
+import org.yarnandtail.andhow.export.PropertyExport;
+
+import java.util.Collection;
```
Comment (alex-kar) [inline andhow-core/src/main/java/org/yarnandtail/andhow/api/StaticPropertyConfiguration.java:5]: I guess those changes were added during development and can be removed as unused.

**Comment by alex-kar** [comment]
@eeverman I think this feature really useful. Basically we have the same way to configure frameworks, but with strong types and validation.
Is there a way to define more then one property file `andhow.properties`? For example if we want to split configurations instead of combine in one property file?
What about `yaml` support?

**Review by alex-kar — APPROVED** [review]

**Comment by eeverman** [comment]
> @eeverman I think this feature really useful. Basically we have the same way to configure frameworks, but with strong types and validation.
> Is there a way to define more then one property file `andhow.properties`? For example if we want to split configurations instead of combine in one property file?

Great question:  Yes.  The Loaders you were working on are related to that.  There are standard loaders that are used by default (they all start with 'Std' in the name), and there are other loaders that can be added.  Additional [PropfileOnClasspathLoader](https://github.com/eeverman/andhow/blob/homepage/andhow-core/src/main/java/org/yarnandtail/andhow/load/PropFileOnClasspathLoader.java) instances could be added to the AndHow configuration to fetch values from additional properties files.  Additionally, the path of those files could just be hardcoded, e.g. 'myApp.properties' or it could be a `StrProp` that would allow the name of the file to be configured.

There isn't a good example of doing that, but I could add one - If you would find it useful, likely others would as well.

> What about `yaml` support?

I've thought about yaml support, and it probably would be good.  One reason I have not added it is that this level of configuration is not hierarchical or structured at all, while yaml supports rich data like arrays, nested data, etc..  It would be easy for people to assume that since yml is supported, they can use all the features of yml.

Fundamentally, AndHow really just supports key:value pairs.  And at some level, every application needs configuration at that level.

If you are interested in yml support, however, a Loader instance is pretty easy to write.

Another Loader I've been thinking about including is an AWS secrets Loader.  That would allow applications running on AWS to load their configuration directly from the AWS secrets store - I think that would have a lot of potential users.

File: andhow-core/src/main/java/org/yarnandtail/andhow/api/StaticPropertyConfiguration.java (hunk 1 above)
Comment (eeverman) [reply andhow-core/src/main/java/org/yarnandtail/andhow/api/StaticPropertyConfiguration.java]: Good catch.  Fixed.

**Pull request merged**